src/WinRT.Interop.Generator/Helpers/GuidGenerator.cs

```diff
@@ -0,0 +1,322 @@
+    /// applied to the specified type.
+    /// </summary>
+    /// <param name="typeDef">The type definition to inspect for the default interface attribute.</param>
+    /// <param name="interopReferences">The <see cref="InteropReferences"/> instance to use.</param>
+    /// <param name="defaultInterfaceSig">When this method returns <c>true</c>, contains the default interface type signature.</param>
+    /// <returns><c>true</c> if the default interface signature was found; otherwise, <c>false</c>.</returns>
+    private static bool TryGetDefaultInterfaceSignatureFromAttribute(TypeDefinition typeDef, InteropReferences interopReferences, [NotNullWhen(true)] out TypeSignature defaultInterfaceSig)
+    {
+        if (typeDef.TryGetCustomAttribute(interopReferences.WindowsRuntimeDefaultInterfaceAttribute, out CustomAttribute? customAttribute))
+        {
+            if (customAttribute.Signature is { FixedArguments: [{ Element: TypeSignature signature }, ..] })
+            {
+                defaultInterfaceSig = signature;
+                return true;
+            }
+        }
+        defaultInterfaceSig = null!;
+        return false;
+    }
+
+    /// <summary>
+    /// Encodes a GUID from a 16-byte sequence following RFC 4122 rules.
+    /// Adjusts byte order for little-endian systems and sets version and variant bits.
+    /// </summary>
+    /// <param name="data">A read-only span of bytes representing the GUID data. Must be at least 16 bytes.</param>
+    /// <returns>A <see cref="Guid"/> constructed from the encoded byte sequence.</returns>
+    /// <exception cref="ArgumentException">Thrown when <paramref name="data"/> contains fewer than 16 bytes.</exception>
+    private static Guid EncodeGuid(ReadOnlySpan<byte> data)
+    {
+        if (data.Length < 16)
+        {
+            throw new ArgumentException("Data must be at least 16 bytes.", nameof(data));
+        }
+
+        Span<byte> buffer = stackalloc byte[16];
+        data[..16].CopyTo(buffer);
+
+        if (BitConverter.IsLittleEndian)
+        {
+            // Swap bytes of int a
+            (buffer[3], buffer[0]) = (buffer[0], buffer[3]);
+            (buffer[2], buffer[1]) = (buffer[1], buffer[2]);
+
+            // Swap bytes of short b
+            (buffer[5], buffer[4]) = (buffer[4], buffer[5]);
+
+            // Swap bytes of short c and encode RFC time/version field
+            (buffer[7], buffer[6]) = ((byte)((buffer[6] & 0x0F) | (5 << 4)), buffer[7]);
+
+            // Encode RFC clock/reserved field
+            buffer[8] = (byte)((buffer[8] & 0x3F) | 0x80);
+        }
+
+        return new Guid(buffer);
+    }
+
+    /// <summary>Computes a deterministic GUID (IID) from a WinRT signature</summary>
+    /// <param name="signature">WinRT signature string.</param>
+    /// <returns>The derived <see cref="Guid"/>.</returns>
+    private static Guid CreateGuidFromSignature(string signature)
```

Comment:
The parameter here could be `ReadOnlySpan<char>`
```suggestion
    private static Guid CreateGuidFromSignature(ReadOnlySpan<char> signature)
```